### PR TITLE
Upgrade mlnx (syncd/syncd-rpc/saiserver) to bookworm

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Upgrade Mellanox (syncd/syncd-rpc/saiserver) containers to Bookworm

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How to verify it

Manual Verification and Regression tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: platform/mellanox/docker-saiserver-mlnx/Dockerfile.j2 [hunk 1]
```diff
@@ -40,7 +40,7 @@ python-debs/{{ deb }}{{' '}}
 {%- endfor -%}
 debs/
 
-RUN apt-get install -y --no-install-recommends libxml2 iptables libbsd0 protobuf-c-compiler protobuf-compiler python3-protobuf libprotobuf-c1 python3-future python3-ipaddr libnet1 pkg-config asciidoc xmlto
```
Comment (dgsudharsan) [inline platform/mellanox/docker-saiserver-mlnx/Dockerfile.j2:43]: Why is pytho3-ipaddr removed?

File: platform/mellanox/mlnx-sai.mk [hunk 2]
```diff
@@ -21,6 +21,7 @@ MLNX_SAI = mlnx-sai_1.mlnx.$(MLNX_SAI_VERSION)_$(CONFIGURED_ARCH).deb
 $(MLNX_SAI)_SRC_PATH = $(PLATFORM_PATH)/mlnx-sai
 $(MLNX_SAI)_DEPENDS += $(MLNX_SDK_DEBS) $(LIBNL_ROUTE3_DEV)
 $(MLNX_SAI)_RDEPENDS += $(MLNX_SDK_RDEBS) $(MLNX_SDK_DEBS) $(LIBNL_ROUTE3)
+
```
Comment (dgsudharsan) [inline platform/mellanox/mlnx-sai.mk:24]: We can remove the diff in this file

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: platform/mellanox/docker-saiserver-mlnx/Dockerfile.j2 (hunk 1 above)
Comment (vivekrnv) [reply platform/mellanox/docker-saiserver-mlnx/Dockerfile.j2:43]: This is deprecated in bookworm https://packages.debian.org/bullseye/python3-ipaddr

Besides, the ipaddress lib is already part of python3.11

```
vkarri@sonic-server:/sonic-buildimage$ docker ps -a
CONTAINER ID   IMAGE                            COMMAND                  CREATED          STATUS                      PORTS     NAMES
dcb26195790d   docker-saiserverv2-mlnx:latest   "/usr/local/bin/supe…"   44 seconds ago   Up 43 seconds                         upbeat_rosalind

vkarri@r-build-sonic06:/builds2/vkarri/sonic/sonic-buildimage$ docker exec -it upbeat_rosalind bash

root@dcb26195790d:/# python3
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ipaddress
>>> 
```

File: platform/mellanox/mlnx-sai.mk (hunk 2 above)
Comment (vivekrnv) [reply platform/mellanox/mlnx-sai.mk]: updated

**Pull request closed without merging**